index.d.ts

```diff
@@ -0,0 +1,267 @@
+import EventEmitter from 'events';
+import { PassThrough, Writable } from 'stream';
+
+type EventsType = {
+  PASSED: 'passed';
+  FAILED: 'failed';
+  SKIPPED: 'skipped';
+  STARTED: 'started';
+  FINISHED: 'finished';
+}
+
+type zUnitOptions = Partial<{
+  defaults: unknown;
+  initial: unknown;
+  runtime: unknown;
+  bequeathed: unknown;
+}>;
+
+declare class Runnable extends EventEmitter.EventEmitter {
+  constructor(name: string);
+
+  name: string;
+
+  description: string;
+}
+
+type zUnitSyntax = {
+  describe: typeof describe;
+  xdescribe: typeof xdescribe;
+  odescribe: typeof odescribe;
+  it: typeof it;
+  xit: typeof xit;
+  oit: typeof oit;
+  before: typeof before;
+  beforeEach: typeof beforeEach;
+  after: typeof after;
+  afterEach: typeof afterEach;
+  include: typeof include;
+}
+
+type OutcomesType = {
+  PASSED: 'passed';
+  FAILED: 'failed';
+  SKIPPED: 'skipped';
+}
+
+export const Events: EventsType;
+
+export class Harness extends EventEmitter.EventEmitter {
+  constructor(testable: Testable, initial?: zUnitOptions);
+
+  numberOfTests: number;
+
+  report: unknown;
+
+  run(reporter: unknown, runtime?: zUnitOptions): Promise<unknown>;
```

Comment:
The reporter must have the following functions...

1. withHarness(harness): harness
1. withSuite(suite): suite
1. withTest(test): test

Reply:
So like this?
```ts
run(reporter: {
  withHarness(harness: Harness): Harness;
  withSuite(suite: Suite): Suite;
  withTest(test: Test): Test;
}, runtime?: zUnitOptions): Promise<unknown>;
```

Reply:
Almost, the `runtime` parameter type isn't zUnitOptions. It's a map like 
```json
{
  timeout: 1000,
}
```

Reply:
Yes, exactly

Reply:
I reckon I have this resolved, I'll submit a new PR today with this and all other changes